test/core/pipeline/features/test_run.py

```diff
@@ -2821,3 +2821,148 @@ def run(self, replies: List[str]):
             )
         ],
     )
+
+@given("a pipeline that passes outputs that are consumed in cycle to outside the cycle", target_fixture="pipeline_data")
+def passes_outputs_outside_cycle():
```

Comment:
@tstadel could you check if this test covers the use case that you had with passing the prompt outside the cycle?

Reply:
The test fails on main and it yields an empty prompt instead of the expected prompt value.

Reply:
The output from the cycle isn't distributed in the old logic because we are filtering for components that are in the cycle here:

https://github.com/deepset-ai/haystack/blob/167ede1f4ca8996aa3fece336ba13fde85724e95/haystack/core/pipeline/pipeline.py#L213

And then we remove any output that was already received by a component in the cycle here: https://github.com/deepset-ai/haystack/blob/167ede1f4ca8996aa3fece336ba13fde85724e95/haystack/core/pipeline/base.py#L889

That means that outputs that are received both by components inside and outside the cycle will only go to components inside the cycle.